pkg/config/engine/containerd/containerd.go

```diff
@@ -36,14 +37,64 @@ var _ engine.Interface = (*Config)(nil)
 
 // New creates a containerd config with the specified options
 func New(opts ...Option) (engine.Interface, error) {
-	b := &builder{}
+	b := &builder{
```

Comment:
Let's rather merge / review the refactoring from #643 as a separate PR so that we can better discuss the additions of this PR.